test/DebugInfo/method.swift

```diff
@@ -0,0 +1,27 @@
+// RUN: %target-swift-frontend -primary-file %s -emit-ir -gdwarf-types -o - | %FileCheck %s
+
+// Verify that we added a declaration for a method.
+public struct Foo {
+    static func static_method() {}
+    func method() {}
+}
+
+public class Bar {
+    static func static_method() {}
+    func method() {}
+}
+
+func a_function() {}
+
+// CHECK: distinct !DISubprogram(name: "static_method",
+// CHECK-SAME: declaration:
+// CHECK: distinct !DISubprogram(name: "method",
+// CHECK-SAME: declaration:
+// CHECK: distinct !DISubprogram(name: "static_method",
+// CHECK-SAME: declaration:
+// CHECK: distinct !DISubprogram(name: "method",
+// CHECK-SAME: declaration:
```

Comment:
Could you also add a CHECK testing that the declaration is indeed for the same method and that the parent type (e.g., Foo) lists the declaration, not the definition?